src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2845,6 +2846,41 @@ private SyntaxToken LexDirectiveToken()
             return Create(in info, null, trailing, errors);
         }
 
+        public SyntaxToken LexEndOfDirectiveWithOptionalPreprocessingMessage()
+        {
+            PooledStringBuilder? builder = null;
+
+            // Skip the rest of the line until we hit a EOL or EOF.  This follows the PP_Message portion of the specification.
+            while (true)
+            {
+                var ch = this.TextWindow.PeekChar();
+                if (ch is '\r' or '\n' || SyntaxFacts.IsNewLine(ch))
+                {
+                    // don't consume EOL characters here
+                    break;
+                }
+                else if (ch is SlidingTextWindow.InvalidCharacter && this.TextWindow.IsReallyAtEnd())
+                {
+                    // don't consume EOF characters here
+                    break;
+                }
+
+                builder ??= PooledStringBuilder.GetInstance();
+                builder.Builder.Append(ch);
+                this.TextWindow.AdvanceChar();
+            }
```

Comment:
new approach is much more in line with the spec.  we do no interpretation of characters.  We simply consume uniformly until we hit EOL or EOF.  All the data (prior to that) is the message portion of hte directive.